scps/SCP-002.md

```diff
@@ -0,0 +1,69 @@
+## SCP-002: Datastore P2P interface
```

Comment:
Right out of the gates I think we're gonna want to look at this one layer of abstraction upward:

[This interface](https://github.com/orbitdb/orbit-db-pubsub)  should be flexible enough to handle the internal needs of the datastores, as well as the external needs of whatever pubsub interface we end up using.

**Internal** requirements are defined as whatever this proposal ends up being, as well as whatever the current functionality is.

**External** means that _eventually_ OrbitDB should be able to use MQTT, IPFS pubsub, whatever else we can think of
https://github.com/orbitdb/orbit-db-pubsub

Reply:
For example one possible external and internal requirement of the interface might be the use of a peerId in the pubsub messages to verify authenticity. However, if the pubsub like system being utilized does not support cryptographically signed peerIds and messages, then it might break certain functionality. I am not sure if MQTT has support for cryptographically signed messages and peerIds. If so we also need to make sure the formatting of those constructs is compatible, or can be transformed into a data type that is indifferent between systems.